app/src/Level2_simple_agentic_with_websearch.ipynb

```diff
@@ -0,0 +1,318 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "base_url = LLAMA_STACK_ENDPOINT if remote else f\"http://localhost:{LLAMA_STACK_ENDPOINT}\"\n",
+    "client = LlamaStackClient(\n",
+    "        base_url=base_url,\n",
+    "        provider_data={\"tavily_search_api_key\": TAVILY_SEARCH_API_KEY}\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Configuring an agent for tool use.\n",
+    "- **Agent Initialization**: \n",
+    "\n",
+    "- Create an `Agent` instance with the desired model, instructions, tools.\"\n",
+    "\n",
+    "- **Instructions**: The `instructions` parameter, also referred to as the system prompt, specifies the agent's role and behavior. In this example, the agent is configured as a helpful web search assistant. It is instructed to use a tool whenever a web search is required and to respond in a friendly and helpful tone.\n",
+    "\n",
+    "- **Tools**: The `tools` parameter defines the tools available to the agent. In this case, the `builtin::websearch` tool is used, which enables the agent to perform web searches. This tool is essential for retrieving up-to-date information from the web. Internally, it leverages Tavily Search to execute the search queries efficiently.\n",
+    "\n",
+    "- **Tool Configuration**: The `tool_config` parameter allows for additional customization of the tools. For instance, the `\"tool_choice\": \"required\"` setting ensures that the agent must invoke the specified tool when performing a task. \n",
+    "\n",
+    "- **How It Works**: When a user query is provided, the agent processes the input and determines whether a tool is required to fulfill the request. If the query involves retrieving information from the web, the agent invokes the `builtin::websearch` tool. The tool interacts with Tavily Search to fetch real-time data, which is then processed and returned to the user in a friendly and helpful tone. This workflow ensures that the agent can handle a wide range of queries effectively.\n",
+    "\n",
+    "For more details on the `builtin::websearch` tool and its capabilities, refer to the [Llama-stack tools documentation](https://llama-stack.readthedocs.io/en/latest/building_applications/tools.html). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "agent = Agent(\n",
+    "    client, \n",
+    "    model=inference_model,\n",
+    "    instructions=\"\"\"You are a helpful websearch assistant. When you are asked to search the web you must use a tool. \n",
+    "            Whenever a tool is called, be sure return the response in a friendly and helpful tone.\n",
+    "            \"\"\" ,\n",
+    "    tools=[\"builtin::websearch\"],\n",
+    "    tool_config={\"tool_choice\":\"required\"},\n",
```

Comment:
If `required` is set here, will it use the websearch tool for every interaction? Even if you just say like "hello"? Does this limit its capacity to have a multi-turn conversation about a single web search?